### PR TITLE
Cython Statistics for SSA Trajectories

### PR DESCRIPTION
**Pull request by WilliamIX**
Merging WP-dev into dev. This branch has a bunch of fast cython functions as part of the SSAResults class that calculate statistics from stochastic simulations:

* empirical_distribution(self, double burn_in, list species_inds, double tend, list max_counts_list) 
* first_moment(self, double start_time, double final_time, list species_inds) 
* standard_deviation(self, double start_time, double final_time, list species_inds)
* second_moment(self, double start_time, double final_time, list species_inds1, list species_inds2)
* correlations(self, double start_time, double final_time, list species_inds1, list species_inds2)

The first of these has been tested extensively. The others less so.

### CODE REVIEW

File: bioscrape/simulator.pyx [hunk 1]
```diff
@@ -621,6 +621,293 @@ cdef class SSAResult:
             return self.py_get_result()
 
 
+    def py_empirical_distribution(self, start_time = None, species = None, Model = None, final_time = None, max_counts = None):
+
```
Comment (ayush9pandey) [inline bioscrape/simulator.pyx:625]: Can you add docstrings here for a general description of what this function can do. I think you already have comments explaining things which could be moved here.

Comment (ayush9pandey) [reply bioscrape/simulator.pyx:625]: and if possible, same for other functions here too.

**Review by ayush9pandey — APPROVED** [review]
All changes look good to me! Looking forward to using these new features. I made a comment about docstrings. In future, you might wanna add a jupyter notebook demonstrating these cool features.

File: bioscrape/simulator.pyx (hunk 1 above)
Comment (ayush9pandey) [reply bioscrape/simulator.pyx:625]: Thanks! Did the commit not go through though? I can't see the new commit..

**Comment by WilliamIX** [comment]
@ayush9pandey just created an examples notebook as well.

**Comment by ayush9pandey** [comment]
Great! Let's merge this once the check is complete.

**Comment by ayush9pandey** [comment]
Looks like some version error with Python 3.7

**Pull request merged**